### DISCUSSION TITLE
Case not created on Voicemail Test

### DISCUSSION THREAD

**Discussion by jlaurie87** [amazon-connect/amazon-connect-salesforce-scv#104] [Q&A] [answered]
I have configured Voicemail Express V2 as in the guide, however no case is made after ending the call. 

I can confirm the Validator Step was a success. 
The logs for packager appear to have an error for description, and I dont see a failed to create salesforce case error, I actually dont see any salesforce logs in the packager log. 

START RequestId: a1b8024f-5c3e-4e47-97d6-cffb184d18d4 Version: $LATEST

[ERROR]	2022-05-24T21:01:10.809Z	a1b8024f-5c3e-4e47-97d6-cffb184d18d4	'Description'
[ERROR]	2022-05-24T21:01:10.809Z	a1b8024f-5c3e-4e47-97d6-cffb184d18d4	Record 1 Result: Failed to extract queue name
[ERROR]	2022-05-24T21:01:10.925Z	a1b8024f-5c3e-4e47-97d6-cffb184d18d4	local variable 'entity_description' referenced before assignment
[ERROR]	2022-05-24T21:01:10.925Z	a1b8024f-5c3e-4e47-97d6-cffb184d18d4	Record 1 Result: Failed to extract attributes
[ERROR] KeyError: 'json_attributes'Traceback (most recent call last):  File "/var/task/vmx_packager.py", line 182, in lambda_handler    if 'vm_mode' in writer_payload['json_attributes']:
END RequestId: a1b8024f-5c3e-4e47-97d6-cffb184d18d4

**Comment by dougjaso** [comment]
I assume this is attempting to route a queue voicemail. There is a bug in the current setup. Your queue needs to have something in the description. Go into Connect. Modify the queue to have a description. Retry.

-----------
Jason Douglas
Principal Solutions Architect
Amazon Connect

628.336.1973 – mobile
415.262.9741 – office

[signature_1544445572]<http://www.aws.amazon.com/>


From: jlaurie87 ***@***.***>
Reply-To: amazon-connect/amazon-connect-salesforce-scv ***@***.***>
Date: Tuesday, May 24, 2022 at 2:18 PM
To: amazon-connect/amazon-connect-salesforce-scv ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [amazon-connect/amazon-connect-salesforce-scv] Case not created on Voicemail Test (Discussion #104)


I have configured Voicemail Express V2 as in the guide, however no case is made after ending the call.

I can confirm the Validator Step was a success.
The logs for packager appear to have an error for description, and I dont see a failed to create salesforce case error, I actually dont see any salesforce logs in the packager log.

START RequestId: a1b8024f-5c3e-4e47-97d6-cffb184d18d4 Version: $LATEST

[ERROR] 2022-05-24T21:01:10.809Z a1b8024f-5c3e-4e47-97d6-cffb184d18d4 'Description'
[ERROR] 2022-05-24T21:01:10.809Z a1b8024f-5c3e-4e47-97d6-cffb184d18d4 Record 1 Result: Failed to extract queue name
[ERROR] 2022-05-24T21:01:10.925Z a1b8024f-5c3e-4e47-97d6-cffb184d18d4 local variable 'entity_description' referenced before assignment
[ERROR] 2022-05-24T21:01:10.925Z a1b8024f-5c3e-4e47-97d6-cffb184d18d4 Record 1 Result: Failed to extract attributes
[ERROR] KeyError: 'json_attributes'Traceback (most recent call last):  File "/var/task/vmx_packager.py", line 182, in lambda_handler    if 'vm_mode' in writer_payload['json_attributes']:
END RequestId: a1b8024f-5c3e-4e47-97d6-cffb184d18d4

**Reply by jlaurie87** [reply]
@dougjaso Yep, I found that and did add the description. 

Solved that issue, but now I get a new error:

START RequestId: 31c305f8-0a57-48e7-9235-b928acfee60b Version: $LATEST

[ERROR]	2022-05-24T21:23:49.603Z	31c305f8-0a57-48e7-9235-b928acfee60b	list indices must be integers or slices, not str
[ERROR]	2022-05-24T21:23:49.603Z	31c305f8-0a57-48e7-9235-b928acfee60b	Record 1 Result: Failed to create Salesforce Case
END RequestId: 31c305f8-0a57-48e7-9235-b928acfee60b

Any ideas what I may have missed?

I think I saw someone else post this same error, but now I cant find the discussion haha.

**Comment by dougjaso** [comment, marked as answer]
What is your current logging level set to? Can you set it to debug and try again? If you have previously passed the Salesforce validation step of sf config, then the most likely issue is that one of the field api names doesn’t match.

Sent from my iPhone

On May 24, 2022, at 2:29 PM, jlaurie87 ***@***.***> wrote:

﻿

@dougjaso<https://github.com/dougjaso> Yep, I found that and did add the description.

Solved that issue, but now I get a new error:

START RequestId: 31c305f8-0a57-48e7-9235-b928acfee60b Version: $LATEST

[ERROR] 2022-05-24T21:23:49.603Z 31c305f8-0a57-48e7-9235-b928acfee60b list indices must be integers or slices, not str
[ERROR] 2022-05-24T21:23:49.603Z 31c305f8-0a57-48e7-9235-b928acfee60b Record 1 Result: Failed to create Salesforce Case
END RequestId: 31c305f8-0a57-48e7-9235-b928acfee60b

Any ideas what I may have missed?

I think I saw someone else post this same error, but now I cant find the discussion haha.

**Reply by jlaurie87** [reply]
OK. I am checking now. I see that the guide says to put the Voicemail_Recording on the custom object, but didnt see it mentioned for Case. 

I have added that and testing now. 

I found the previous reference to this issue, and saw the Voicemail link field missing issue.

**Reply by jlaurie87** [reply]
@dougjaso 

Ok That fixed it. Thank you. 

Q. is there any way to change the Subject to be custom and is there a way to tie an account on a lambda lookup?

**Comment by dougjaso** [comment]
With cases, the transcript is in the description of the case.

Sent from my iPhone

On May 24, 2022, at 2:47 PM, jlaurie87 ***@***.***> wrote:

﻿

OK. I am checking now. I see that the guide says to put the Voicemail_Recording on the custom object, but didnt see it mentioned for Case.

I have added that and testing now.

I found the previous reference to this issue, and saw the Voicemail link field missing issue.

**Comment by dougjaso** [comment]
Sure. You can do both by modifying the sub_sfcase.py code. I’m getting on a plane right now, so can’t look at it at this moment. But you can change the code to do whatever you like. If you’re trying to link to a SF account, it might be easier to do that lookup in your main flow, write that as an attribute, then mod the code to use that attribute however you want.

Sent from my iPhone

On May 24, 2022, at 2:48 PM, jlaurie87 ***@***.***> wrote:

﻿

@dougjaso<https://github.com/dougjaso>

Ok That fixed it. Thank you.

Q. is there any way to change the Subject to be custom and is there a way to tie an account on a lambda lookup?
